src/Umbraco.Web/UmbracoModule.cs

```diff
@@ -323,6 +323,14 @@ internal static bool HandleHttpResponseStatus(HttpContextBase context, Published
             LogHelper.Debug<UmbracoModule>("Response status: Redirect={0}, Is404={1}, StatusCode={2}",
                 () => pcr.IsRedirect ? (pcr.IsRedirectPermanent ? "permanent" : "redirect") : "none",
                 () => pcr.Is404 ? "true" : "false", () => pcr.ResponseStatusCode);
+            
+            response.Cache.SetCacheability(pcr.Cacheability);
+
+            foreach (var cacheExtension in pcr.CacheExtensions)
+                response.Cache.AppendCacheExtension(cacheExtension);
+
+            foreach (var header in pcr.Headers)
+                response.Cache.AppendCacheExtension(string.Format("{0}, {1}", header.Key, header.Value));

```

Comment:
This needs to be `response.AppendHeader` instead of response.Cache.AppendCacheExtension